nemo/collections/llm/recipes/llama3_70b.py

```diff
@@ -93,6 +95,26 @@ def pretrain_recipe(
     )
 
 
+def pretrain_recipe_performance(
+    name: str, ckpt_dir: str, num_nodes: int, num_gpus_per_node: int, fn: Callable = pretrain
+) -> Partial:
+    recipe = pretrain_recipe(
+        name=name, ckpt_dir=ckpt_dir, num_nodes=num_nodes, num_gpus_per_node=num_gpus_per_node, fn=fn
+    )
+
+    recipe.trainer.callbacks.append(
+        Config(
+            MegatronCommOverlapCallback,
+            tp_comm_overlap=True,
+            tp_comm_overlap_cfg=userbuffers_bf16_h100_h8192_tp4_mbs1_seqlen8192,
```

Comment:
Can we wrap this in a getter method based on 
 - precision, hardware , seqlen, batch-size, TP etc
 
np: (optional) runtime detect the hardware or please add a TODO.

Reply:
Something like:
`get_tp_comm_overlap_config(precision, hardware, seqlen, etc) ?